modules/platform/nuxeo-platform-web-common/src/main/java/org/nuxeo/ecm/platform/ui/web/auth/NuxeoAuthenticationFilter.java

```diff
@@ -1142,4 +1146,24 @@ protected Principal getPrincipalCheckingAuth(UserIdentificationInfo userIdent, H
         }
     }
 
+    /**
+     * Checks if the {@link #REQUESTED_URL} request parameter is an absolute URL, in which case, throws a NuxeoException
+     * with a 400 status code.
+     *
+     * @since 11.5
+     */
+    protected void checkRequestedURL(ServletRequest request) {
+        String requestedURL = request.getParameter(REQUESTED_URL);
+        if (requestedURL == null) {
+            return;
+        }
+        try {
+            if (URI.create(requestedURL).isAbsolute()) {
+                throw new NuxeoException(SC_BAD_REQUEST);
+            }
+        } catch (IllegalArgumentException e) {
+            // malformed URI, the requested URL is not absolute
```

Comment:
If the URL cannot be parsed I'd be conservative and throw `new NuxeoException(SC_BAD_REQUEST)` too.